react/week3/to_do_list_with_api/src/index.js

```diff
@@ -0,0 +1,209 @@
+  }
+
+  render() {
+    return (
+      <div id='counterCointainer'>
+        <h3>You have used {this.state.ellapsedTime} seconds</h3>
+      </div>
+    )
+  }
+}
+
+class SingleTaskMaker extends React.Component {
+  state = { ...this.props.task, text: "Edit" }
+
+  toggleDoneStatus = () => {
+    this.setState({ ...this.state, donestatus: !this.state.donestatus })
+  }
+
+  deleteTaks = () => {
+    this.setState({ ...this.state, display: !this.state.display })
+  }
+
+  editTask = () => {
+    if (this.state.editmode) {
+      this.setState({ ...this.state, text: "Edit", editmode: false })
+    } else {
+      this.setState({ ...this.state, text: "Update", editmode: true })
+    }
+  }
+
+  updateTaskDescription = (evnt) => {
+    this.setState({ description: evnt.target.value })
+  }
+
+  taskTitle = () => {
+    if (this.state.editmode) {
+      return (
+        <li className='taskName'>
+          <input type="text" defaultValue={this.state.description} onChange={this.updateTaskDescription} name='taskDescription' ></input>
+        </li>
+      )
+    } else {
+      return (
+        <li
+          className='taskName'
+          key={this.state.id}
+          style={{ textDecoration: this.state.donestatus ? 'line-through' : 'none' }}>
+          {`${this.state.description} | ${this.state.deadline}`}
+        </li>
+      )
+    }
+  }
+
+  render() {
+    if (!this.state.display) return (<></>);
+    return (
+      <li key={'task' + this.state.id} className='taskLine'>
+        <ul>
+          {this.taskTitle()}
+          <input type="checkbox" onChange={this.toggleDoneStatus}></input>
+          <button onClick={this.deleteTaks} > Delete </button>
+
+          <button onClick={e => this.editTask(this.state.editmode)} > {this.state.text} </button>
+
+          <li>{`${new Date().getHours()}:${new Date().getMinutes()}:${new Date().getSeconds()}`}</li>
+        </ul>
+      </li>
+    )
+  }
+
+}
+
+class List extends React.Component {
+  constructor(props) {
+    super(props);
+    this.myRef = React.createRef();
+    this.state = {
+      ...this.props, newTaskDate: new Date(),
+      NewTaskName: 'Add new task',
+    }
+  }
+
+  cleanDefaultValue = () => {
+    this.setState({ NewTaskName: '' })
+  }
+
+  onChangeNewTaskDescription = (evnt) => {
+    this.setState({ NewTaskName: evnt.target.value })
+    console.log(this.state)
+  }
+
+  onChange = date => this.setState({ newTaskDate: date })
+
+  addbuttonClick = () => {
+    console.log(this.state.toDoList);
+    const toDoListArray = [{
+      id: 6,
+      description: this.state.NewTaskName,
+      deadline: '2019-07-14',
+      donestatus: false,
+      display: true,
+      editmode: false,
+    }]
+    // ####################### Question to mentor  #############################
+    // How may I reset the placeholder of the text box 
+    this.setState({ toDoList: this.state.toDoList.concat(toDoListArray), NewTaskName: 'Add new task' })
+  }
+
+  newTaskMaker = () => {
```

Comment:
```suggestion
  createNewTodo = () => {
```
I think this would be a more clear name!

Reply:
Fair enough. I will implement it. Thank you for your advice. :)